src/devices/bus/a2bus/grapplerplus.cpp

```diff
@@ -0,0 +1,236 @@
+// license:BSD-3-Clause
+// copyright-holders:R. Belmont, Golden Child
+/*********************************************************************
+
+    grapplerplus.c
+
+    Orange Micro Grappler Plus Apple II Parallel Interface Card
+
+*********************************************************************/
+
+#include "emu.h"
+#include "grapplerplus.h"
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(A2BUS_GRAPPLERPLUS, a2bus_grapplerplus_device, "grapplerplus", "Grappler Interface Card")
+
+#define GRAPPLERPLUS_ROM_REGION  "grapplerplus_rom"
+#define GRAPPLERPLUS_CENTRONICS_TAG "centronics"
```

Comment:
You shouldn’t need a macro for the device tag – you can pass device finders in place of tags in machine configuration and address maps.

Reply:
I removed the macros and put the strings in directly.  If that's not good, can you point to an example of a device finder.  (I still don't know all the ins and outs of devices).